### PR TITLE
15 Coastal Data Read 

### PR DESCRIPTION
**Pull request by jhrehanoaa**
(ignore the branch name) 

This incorporates the reading of ncdf and txt data from coastal. 

- Reads in two new files from 

- Custominput.yaml produces two dataframe depending on what coastal provides 

- Uses checks in place when they are not provided.

### CODE REVIEW

**Comment by jameshalgren** [comment]
@jhrehanoaa: instead of building this as a complete new input file, could we structure this as a unit test demonstrating that the input file is properly read...?

**Comment by jhrehanoaa** [comment]
I also left out the sandy sample file (which will most likely be the file she passes to us). As it is too large for github.

File: src/python_framework_v02/troute/nhd_network_utilities_v02.py [hunk 1]
```diff
@@ -603,3 +603,14 @@ def build_data_assimilation_folder(data_assimilation_parameters):
...
 
     return usgs_df
+
+
+def build_coastal_dataframe(coastal_output):
+    coastal_df = pd.read_csv(coastal_output, sep="  ", header=None)
+    return coastal_df
+
+
+def build_coastal_ncdf_dataframe(coastal_ncdf):
```
Comment (jameshalgren) [inline src/python_framework_v02/troute/nhd_network_utilities_v02.py:613]: Duplicated in nhd_io

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py [hunk 2]
```diff
@@ -1192,6 +1197,18 @@ def main():
     else:
         usgs_df = pd.DataFrame()
 
+    # STEP 7
+    coastal_output = coastal_parameters.get("coastal_output_data", None)
+    coastal_ncdf = coastal_parameters.get("coastal_ncdf", None)
+
+    if coastal_output:
```
Comment (jameshalgren) [inline src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:1204]: - [ ] Move the function to nhd_io and change the call.

**Review by jameshalgren — APPROVED** [review]

**Pull request merged**